src/pymovements/utils/archives.py

```diff
@@ -172,13 +199,32 @@ def _extract_zip(
         Path to the directory the file will be extracted to.
     compression: str | None
         Compression filename suffix.
+    resume: bool
+        Resume if archive was already previous extracted.
+    verbose: int
+        Print messages for resuming each dataset resource.
     """
     compression_id = _ZIP_COMPRESSION_MAP[compression] if compression else zipfile.ZIP_STORED
     with zipfile.ZipFile(source_path, 'r', compression=compression_id) as archive:
-        archive.extractall(destination_path)
-
-
-_ARCHIVE_EXTRACTORS: dict[str, Callable[[Path, Path, str | None], None]] = {
+        for member in tqdm(archive.filelist):
+            member_filename = member.filename
+            member_dest_path = os.path.join(destination_path, member_filename)
+            if resume:
+                member_size = member.file_size
+                if (
+                    os.path.exists(member_dest_path) and
+                    member_filename[-4:] in _ARCHIVE_EXTRACTORS and
+                    member_size == os.path.getsize(member_dest_path)
+                ):
+                    if verbose:
+                        print(f'Skipping {member_filename} due to previous extraction')
+                    continue
+                archive.extract(member_filename, destination_path)
+            else:
```

Comment:
in contrast to the tar implementation, there is no need for an additional if-else